### PR TITLE
fix: data spreadsheet object drag drop

### PR DESCRIPTION
**Pull request by AustinGitHub**
Closes # https://github.com/carbon-design-system/ibm-products/issues/5799

#### What did you change?

I added a check for if the type of header is an object and to check for specific key property set to true `headerKey` to pick up this text value as the header value otherwise in depth json.stringify does not work properly 

#### How did you test and verify your work?
Storybook

https://github.com/user-attachments/assets/4c2f8abe-d3c2-48c4-878a-c4a9b1eb6523

### CODE REVIEW

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 1]
```diff
@@ -295,11 +295,40 @@ export let DataSpreadsheet = React.forwardRef(
       [cellEditorValue, onDataUpdate]
     );
 
+    const getHeaderKeyText = useCallback((element: any) => {
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:298]: If this function just tries to get the text of a given node, we have a utility for that which I think would make most sense to use here, `getNodeTextContent` (`packages/ibm-products/src/global/js/utils/getNodeTextContent`).

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 2]
```diff
@@ -295,11 +295,40 @@ export let DataSpreadsheet = React.forwardRef(
...
+      if (!element || typeof element !== 'object') {
+        return null;
+      }
+
+      if (element.type === 'span' && element.props.headerKey === true) {
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:303]: How come you check explicitly for `span` here?

**Comment by AustinGitHub** [comment]
Thanks, I will address the comments today

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 1 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:298]: Hi Matthew, I took a look at this and this returns back the entire text content but in our case we have a new line like this where the header name is actually `COLUMN_1` and below it is sort of like a subheader we provide to it but in the context of wanting to know the header text it does not matter to us. 

Screenshot 2024-08-02 at 1 32 00 PM

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 2 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:303]: I was following how we had our setup with, will change this out so it isn't specific, thanks!

**Comment by AustinGitHub** [comment]
Hi @matthewgallo is there any update on this? Thanks!

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 3]
```diff
@@ -295,11 +295,40 @@ export let DataSpreadsheet = React.forwardRef(
       [cellEditorValue, onDataUpdate]
     );
 
+    const getHeaderKeyText = useCallback((element: any) => {
+      if (!element || typeof element !== 'object') {
+        return null;
+      }
+
+      if (element.props.headerKey === true) {
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:303]:
```suggestion
      if (element.props.headerKey) {
```

Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx]: fixed, thanks!

**Comment by AustinGitHub** [comment]
Screenshot 2024-08-14 at 9 37 10 AM

With provided feedback. Also noting `column_name` is not required. I don't see anywhere in code to really mention this as even things like `placement` or `sticky` on the columns properties aren't fully defined somewhere so I think implementation for this is ok.

I also added safety net so if user didn't define `column_name` but the header is an object, we will use the `getNodeTextContent` function.

**Comment by AustinGitHub** [comment]
We have 1 more week for our deadline on our project, would love to be able to get these in and a version released with them if possible. That way we can implement them on our UI as soon as possible, thanks!

**Review by matthewgallo — APPROVED** [review]

**Comment by AustinGitHub** [comment]
Hi @annawen1 can you review when you get the chance? Thanks!

**Review by davidmenendez — APPROVED** [review]

**Comment by AustinGitHub** [comment]
@davidmenendez can you add this back to merge queue please? Thanks!

**Comment by AustinGitHub** [comment]
Is there anything left for this? Thanks!

**Comment by AustinGitHub** [comment]
Not sure why tests are failing, they passed before and nothing major changed :(

**Comment by matthewgallo** [comment]
We're adding support for visual regression testing that isn't fully configured yet. Everything looks good here, we can merge this now. Thanks for you patience with this one Austin!

**Comment by AustinGitHub** [comment]
Thanks!

**Pull request merged**